src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -36,9 +45,30 @@ public Task BeforeStartAsync(DistributedApplicationModel model, CancellationToke
             AddDashboardResource(model);
         }
 
+        _dashboardLogsTask = WatchDashboardLogsAsync();
+
         return Task.CompletedTask;
     }
 
+    public async ValueTask DisposeAsync()
+    {
+        _shutdownTcs.Cancel();
+
+        if (_dashboardLogsTask is not null)
+        {
+            try
+            {
+                await _dashboardLogsTask.ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                distributedApplicationLogger.LogError(ex, "Unexpected error while watching dashboard logs.");
```

Comment:
LogCritical?

Reply:
Should it be? @eerhardt left it at warning.

Reply:
I guess if we are in dispose we are going down anyway.